### PR TITLE
Fix reducer error

### PR DESCRIPTION
**Pull request by Graywing13**
- Can't both return a value and modify state at the same time.

### CODE REVIEW

File: client/src/redux/reducers/viewSlice.js [hunk 1]
```diff
@@ -13,6 +13,18 @@ const _setActiveTripId = (state, action) => {
   _setAppView(state, { payload: AppView.TRIP_OVERVIEW })
 }
 
+const _toggleSidebar = (state) => {
+  state.isSidebarOpen = !state.isSidebarOpen
+}
+
+const _openSidebar = (state) => {
+  state.isSidebarOpen = true
```
Comment (vee-16) [inline client/src/redux/reducers/viewSlice.js:21]: is it okay to be directly assigning state or should we make immutable updates?

cross checking with: https://redux.js.org/tutorials/fundamentals/part-3-state-actions-reducers#rules-of-reducers

Comment (vee-16) [reply client/src/redux/reducers/viewSlice.js:21]: Doc suggests approaching as below, by creating copy and updating its value: 

Screenshot 2023-05-27 at 10 36 42 PM

Comment (NLmeng) [reply client/src/redux/reducers/viewSlice.js:21]: that is traditional redux. we are using redux toolkit. https://redux-toolkit.js.org/usage/immer-reducers
`Redux Toolkit's [createReducer](https://redux-toolkit.js.org/api/createReducer) and [createSlice](https://redux-toolkit.js.org/api/createSlice) automatically use [Immer](https://immerjs.github.io/immer/) internally to let you write simpler immutable update logic using "mutating" syntax. This helps simplify most reducer implementations.`

Comment (NLmeng) [reply client/src/redux/reducers/viewSlice.js:21]: it allows you to write code that appears to mutate the state directly but it is actually applying the updates without mutating

**Review by NLmeng — APPROVED** [review]
lgtm

File: client/src/redux/reducers/viewSlice.js (hunk 1 above)
Comment (vee-16) [reply client/src/redux/reducers/viewSlice.js:21]: Okay, that's awesome!

**Pull request merged**